gobblin-yarn/src/main/java/org/apache/gobblin/yarn/event/ContainerReleaseRequest.java

```diff
@@ -21,26 +21,33 @@
 
 import org.apache.hadoop.yarn.api.records.Container;
 
+import lombok.Getter;
+
 
 /**
  * A type of event for container release requests to be used with a {@link com.google.common.eventbus.EventBus}.
  * This event is different than {@link ContainerShutdownRequest} because it releases the container through
  * the Resource Manager, while {@link ContainerShutdownRequest} shuts down a container through the
  * Node Manager
  */
+@Getter
 public class ContainerReleaseRequest {
+  /**
+   * -- GETTER --
+   *  Get the IDs of the containers to release.
+   *
+   * @return the IDs of the containers to release
+   */
   private final Collection<Container> containers;
+  private final boolean spinUpReplacementContainers;
 
   public ContainerReleaseRequest(Collection<Container> containers) {
     this.containers = containers;
+    this.spinUpReplacementContainers = false;
   }
 
-  /**
-   * Get the IDs of the containers to release.
-   *
-   * @return the IDs of the containers to release
-   */
-  public Collection<Container> getContainers() {
-    return this.containers;
+  public ContainerReleaseRequest(Collection<Container> containers, boolean spinUpReplacementContainers) {
```

Comment:
There is a lombok allargsconstructor . That can be explored for boilerplate code.

A general comment that is suitable across is, declare a variable local or class or method parameter, always as final unless you have a necessary reason not to.